File: BottomNav/app/src/main/java/com/example/bottomnav/MainActivity.java
package com.example.bottomnav;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.navigation.NavigationBarView;


public class MainActivity extends AppCompatActivity {

    private final FragmentHandler fragmentHandler = new FragmentHandler();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // アクティビティ生成が初回の場合は、ホーム画面に遷移
        if (savedInstanceState == null) {
            fragmentHandler.setFragmentManager(getSupportFragmentManager());
            fragmentHandler.changeFragment(R.id.home);
        }
    }

    @Override
    protected void onStart(){
        super.onStart();

        // 画面最下部のボタン表示
        BottomNavigationView bottomNavigationView = findViewById(R.id.nav);
        bottomNavigationView.setOnItemSelectedListener(new NavigationBarView.OnItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                return fragmentHandler.changeFragment(item.getItemId());
            }

        });
    }

    @Override
    public void onBackPressed() {
    }
}

File: BottomNavigationApp-main/app/src/main/java/com/example/navigationapplication/MainActivity.kt
package com.example.navigationapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import androidx.core.view.size
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() ,BottomNavigationView.OnNavigationItemSelectedListener{


	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)

		initializeResource()
	}

	fun initializeResource(){
//		BottomNavigationViewを設定
		val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottom_navigation_view)
		bottomNavigationView.setOnNavigationItemSelectedListener(this)
		bottomNavigationView.itemIconSize = 70
		bottomNavigationView.scaleX = 1.2f
		bottomNavigationView.scaleY = 1.2f

//		初期Fragmentを設定
		supportFragmentManager.beginTransaction()
				.replace(R.id.container,HomeFragment())
				.setReorderingAllowed(true)
				.commit()
	}

	override fun onNavigationItemSelected(item: MenuItem): Boolean {
		Log.d(TAG,"Selected item: " + item)

		when(item.itemId){
//			ホームボタンが押された時
			R.id.action_register_clothes -> {
				supportFragmentManager.beginTransaction()
						.replace(R.id.container,HomeFragment())
						.setReorderingAllowed(true)
						.commit()
			}
//			設定ボタンが押された時
			R.id.action_closet -> {
				supportFragmentManager.beginTransaction()
						.replace(R.id.container,SettingFragment())
						.setReorderingAllowed(true)
						.commit()
			}

		}

		return true
	}

	companion object{
		const val TAG : String = "MainActivity"
	}
}